tfutils/base.py

```diff
@@ -184,26 +233,35 @@ def load_from_db(self, query, cache_model=False, force_fetch=False):
         Args:
             query: dict expressing MongoDB query
         """
+        if collfs is None:
+            collfs = self.collfs
+        coll = collfs._GridFS__files
+        if collfs_recent is None:
+            collfs_recent = self.collfs_recent
+        coll_recent = collfs_recent._GridFS__files
+        
         query['saved_filters'] = True
-        count = self.collfs.find(query).count()
+        count = collfs.find(query).count()
         if count > 0:  # get latest that matches query
-            ckpt_record = self.coll.find(query,
-                                         sort=[('uploadDate', -1)])[0]
-            loading_from = self.coll
+            ckpt_record = coll.find(query, sort=[('uploadDate', -1)])[0]
+            loading_from = coll
+        else:
+            ckpt_record = None
 
-        count_recent = self.collfs_recent.find(query).count()
+        count_recent = collfs_recent.find(query).count()
         if count_recent > 0:  # get latest that matches query
-            ckpt_record_recent = self.coll_recent.find(query,
-                                                       sort=[('uploadDate', -1)])[0]
+            ckpt_record_recent = coll_recent.find(query,
+                                            sort=[('uploadDate', -1)])[0]
             # use the record with latest timestamp
             if ckpt_record is None or ckpt_record_recent['uploadDate'] > ckpt_record['uploadDate']:
-                loading_from = self.coll_recent
+                loading_from = coll_recent
                 ckpt_record = ckpt_record_recent
 
         if count + count_recent == 0:  # no matches for query
             log.warning('No matching checkpoint for query "{}"'.format(repr(query)))
             return
 
+        database = loading_from._Collection__database
```

Comment:
No because if count + count_recent > 0 then loading_from will not be undefined, since otherwise, the function will return beforehand.